pdf/core/parser.go

```diff
@@ -1109,8 +1109,15 @@ func (parser *PdfParser) parseXref() (*PdfObjectDictionary, error) {
 	var err error
 	var trailerDict *PdfObjectDictionary
 
+	// Peek 20 bytes to the left and 20 bytes to the right of
+	// the current offset. Reset to the original offset after reading.
+	const tolerance = 20
+	offset := parser.GetFileOffset()
+	parser.SetFileOffset(offset - tolerance)
```

Comment:
Might be better to only try this if the first attempt fails.
Also note that calling SetFileOffset many times can be a bit expensive as a new buffered reader is created every time.  Might be possible to improve that in #442 as creating a new buffer might not be worth it when seeking a short distance.